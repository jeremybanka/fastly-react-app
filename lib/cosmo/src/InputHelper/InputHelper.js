/* @flow */

import * as React from "react";
import { FormattedMessage } from "react-intl";
import { Text } from "../";

type TextType =
  | string
  | React.Element<typeof Text>
  | React.Element<typeof FormattedMessage>;

type InputHelperProps = {
  htmlFor: string,
  children: TextType,
};
const InputHelper = (props: InputHelperProps) => {
  return (
    <Text as="label" htmlFor={props.htmlFor} color="grays.4" fontSize="xxs">
      {props.children}
    </Text>
  );
};

export default InputHelper;
