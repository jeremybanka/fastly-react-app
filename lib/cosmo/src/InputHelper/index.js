/* @flow */

import InputHelper from "./InputHelper";

export default InputHelper;
