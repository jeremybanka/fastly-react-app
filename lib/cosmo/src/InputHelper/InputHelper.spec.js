/* @flow */

import React from "react";
import { render } from "enzyme";
import InputHelper from "./InputHelper";

describe("InputHelper test suite", () => {
  const component = <InputHelper>Helper</InputHelper>;
  it("renders the stored snapshot", () => {
    expect(render(component)).toMatchSnapshot();
  });
});
