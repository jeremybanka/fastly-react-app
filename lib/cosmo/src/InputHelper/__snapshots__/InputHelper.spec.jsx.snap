// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputHelper test suite renders the stored snapshot 1`] = `
<label
  class="Text-sc-6c1xd4-0 bDXzpq"
  color="grays.4"
  font-size="xxs"
>
  Helper
</label>
`;
